src/components/Code/samples/toggle.tsx

```diff
@@ -0,0 +1,41 @@
+import { mdx } from "@mdx-js/react";
+import { send, useContext } from "../../Page/context";
+
+export default function Sample({ language, children }) {
```

Comment:
I might not understand this fully, but to me it seems like there is a lot of code outside of MDX needed in order to get this running, which is a pity. _If_ it's already in MDX, couldn't more be done there?

```
Here's some MDX text where you can <button onClick="send" target="sample-1" value="haskell">toggle Haskell</button> code.

<Code data-id="sample-1" forward={{value: "language"}}>
  <MyCodeExample {/* language will be automatically forwarded */} />
</Code>
```

Not a nice example, I know, I invented it on the spot 😅 But in this example there would be no need to access the message bus directly, which I think would be beneficial, because the components will not be tied to Timvir.

Reply:
You're right, there is currently a lot of code outside of MDX. There are many options how to make the integration into MDX more convenient. I'm a bit hesitant to explore that right now because whatever that will be, it will be part of the public API and I'm not ready yet to commit to that. Not until I collect some more use cases and have a better idea what the requirements are.